inc/WeatherWidget.h

```diff
@@ -0,0 +1,106 @@
+#ifndef WEATHER_WIDGET
+#define WEATHER_WIDGET
+
+#include <string>
+#include "Widget.h"
+#include <raylib.h>
+#include <iostream>
+#include <ctime>
+#include "Constants.h"
+#include "Utility.h"
+#include <json/json.h>
+#include <sstream>
+#include <map>
+using namespace std;
+
+class WeatherWidget : public Widget {
+
+public:
+    /* Docker Checklist
+            Open CMD
+            cd to Smart-Mirror Directory
+            find ip with ipconfig and replace <ip_address> with it
+            run docker run --rm -i -t -v %cd%:/usr/smart-mirror -w /usr/smart-mirror -e DISPLAY=<ip_address>:0.0 smart-mirror
+            you're good to go*/
+    WeatherWidget(uint32_t x, uint32_t y, uint32_t w, uint32_t h, std::string n) :
+        Widget (x, y, w, h, n) { 
+
+        Image weatherIcons = LoadImage("../resources/Adjusted-Weather-Icons.png");
+        ImageResize(&weatherIcons, imageSizeX, imageSizeY);
+        icons = LoadTextureFromImage(weatherIcons);
+        UnloadImage(weatherIcons);
+
+        std::ostringstream api_url;
+        api_url << "https://api.openweathermap.org/data/2.5/weather?lat="
+         << std::to_string(AUSTIN_LATITUDE) << "&lon=" << std::to_string(AUSTIN_LONGITUDE) 
+         << "&appid=" << WEATHER_API;
+        get_http_request(&weatherAPIFinished, api_url.str(), &informations);
+    };
+
+    void draw() override {
+        std::string temp;
+        std::string weather;
+        float ktemp = -1;
+        float ctemp = -1;
+        float ftemp = -1;
+
+        if (weatherAPIFinished) {
+            // Use json here
+            Json::StreamWriterBuilder builder;
+            builder["indentation"] = ""; 
+            // Convert json to string
+            ktemp = stof(Json::writeString(builder, informations["main"]["temp"]));
+            weather = Json::writeString(builder, informations["weather"][0]["main"]);
+            weather = weather.substr(1, weather.length() - 2);
+            // Convert to celsius and fahrenheit
+            ctemp = ktemp - 273.15;
+            ftemp = ((ctemp * 9)/5) + 32;
+            std::ostringstream x;
+            x << std::to_string((int) (isCelsius ? ctemp : ftemp)) << (isCelsius ? "°C" : "°F");
+            temp = x.str();
+        }
+        else {
+            // don't use json here!
+            temp = "Loading...";
+            weather = "Loading...";
+        }
+        // Obtains rectangle for weather icon
+        Rectangle rect = iconMap[weather];
+        // Position for icon on screen
+        Vector2 icon_pos {pos_x, pos_y};
+        DrawTextureRec(icons, rect, icon_pos, WHITE);
+
+        DrawText(temp.c_str(), pos_x + 80, pos_y + 30, 30, SKYBLUE); 
+        DrawText(weather.c_str(), pos_x, pos_y + 70, 30, SKYBLUE); 
+
+    }
+
+private:
+    Texture2D icons;
+    bool weatherAPIFinished = false;
+    bool isCelsius = false;
+    Json::Value informations = new Json::Value();
+    int imageSizeX = MONITOR_SIZE_X / 2;
```

Comment:
If you aren't going to change these values, set them as constant with proper capitalization.